dask_cloudprovider/azure/azurevm.py

```diff
@@ -478,6 +479,7 @@ def __init__(
             "azurevm.public_ingress", override_with=public_ingress
         )
         self.credentials, self.subscription_id = get_azure_cli_credentials()
+        self.credentials = DefaultAzureCredential()
```

Comment:
Perhaps we could add a check for none here?

```suggestion
        if self.credentials is None:
            self.credentials = DefaultAzureCredential()
```

Reply:
There was a slighly different problem related to the internal AZ logic change: https://docs.microsoft.com/en-us/cli/azure/msal-based-azure-cli

This should be more generic approach for receiving credentials and subscription_id